src/main/java/org/terasology/web/webSocket/ClientMessage.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.webSocket;
+
+import com.google.gson.JsonElement;
+
+public class ClientMessage {
+
+    public enum MessageType {
+        AUTHENTICATION_REQUEST,
+        AUTHENTICATION_DATA,
+        RESOURCE
+    }
+
+    private MessageType messageType;
+    private JsonElement data;
+
+    public MessageType getMessageType() {
+        return messageType;
+    }
+
+    public JsonElement getData() {
+        return data;
+    }
+
+    public void checkValid() throws InvalidClientMessageException {
```

Comment:
I think that it would be better to perform the sanity checks in the constructor instead of hoping that the caller will invoke this method.

You can do that by subclassing [InstanceCreator](https://static.javadoc.io/com.google.code.gson/gson/2.6.2/com/google/gson/InstanceCreator.html) and registering it.